qemu/tests/qemu_guest_agent_snapshot.py

```diff
@@ -55,18 +55,25 @@ def _action_after_fsthaw(self, *args):
         if self.bg:
             self.bg.join()
         # Make sure the returned file is identical to the original one
-        self.host_path_returned = "%s-returned" % self.host_path
-        self.vm.copy_files_from(self.guest_path, self.host_path_returned)
-        error_context.context("comparing hashes", logging.info)
-        self.curr_hash = crypto.hash_file(self.host_path_returned)
-        if self.orig_hash != self.curr_hash:
-            self.test.fail("Current file hash (%s) differs from "
-                           "original one (%s)" % (self.curr_hash,
-                                                  self.orig_hash))
-
-        error_context.context("Reboot and shutdown guest.")
-        self.vm.reboot()
-        self.vm.destroy()
+        try:
+            self.host_path_returned = "%s-returned" % self.host_path
+            self.vm.copy_files_from(self.guest_path, self.host_path_returned)
+            error_context.context("comparing hashes", logging.info)
+            self.curr_hash = crypto.hash_file(self.host_path_returned)
+            if self.orig_hash != self.curr_hash:
+                self.test.fail("Current file hash (%s) differs from "
+                               "original one (%s)" % (self.curr_hash,
+                                                      self.orig_hash))
+        finally:
+            error_context.context("Delete the created files.", logging.info)
+            process.run("rm -rf %s %s" % (self.host_path, self.host_path_returned))
+            session = self._get_session(self.params, None)
+            self._open_session_list.append(session)
+            cmd_del_file = "rm -rf %s" % self.guest_path
+            if self.params.get("os_type") == "windows":
+                cmd_del_file = r"del /f /q %s" % self.guest_path
+            session.cmd(cmd_del_file)
+            session.close()
```

Comment:
If you put the `session` into `self._open_session_list`, then you don't need to close it on line 76.
Or you don't put it into the list.

Reply:
You are right, delete it in L76. Pls have a check,thanks.